openmc/tallies.py

```diff
@@ -1076,48 +1031,44 @@ def get_filter_indices(self, filters=[], filter_bins=[]):
         cv.check_type('filters', filters, Iterable, openmc.FilterMeta)
         cv.check_type('filter_bins', filter_bins, Iterable, tuple)
 
-        # Determine the score indices from any of the requested scores
-        if filters:
-            # Initialize empty list of indices for each bin in each Filter
-            filter_indices = []
-
-            # Loop over all of the Tally's Filters
-            for i, self_filter in enumerate(self.filters):
-                # If a user-requested Filter, get the user-requested bins
-                for j, test_filter in enumerate(filters):
-                    if type(self_filter) is test_filter:
-                        bins = filter_bins[j]
-                        break
-                else:
-                    # If not a user-requested Filter, get all bins
-                    if isinstance(self_filter, openmc.DistribcellFilter):
-                        # Create list of cell instance IDs for distribcell Filters
-                        bins = list(range(self_filter.num_bins))
+        # If user did not specify any specific Filters, use them all
+        if not filters:
+            return np.arange(self.num_filter_bins)
 
-                    elif isinstance(self_filter, openmc.EnergyFunctionFilter):
-                        # EnergyFunctionFilters don't have bins so just add a None
-                        bins = [None]
+        # Initialize empty list of indices for each bin in each Filter
+        filter_indices = []
 
-                    else:
-                        # Create list of IDs for bins for all other filter types
-                        bins = self_filter.bins
+        # Loop over all of the Tally's Filters
+        for i, self_filter in enumerate(self.filters):
+            # If a user-requested Filter, get the user-requested bins
+            for j, test_filter in enumerate(filters):
+                if type(self_filter) is test_filter:
```

Comment:
Since we're here, any reason not to use 
```suggestion
                if isinstance(self_filter, test_filter):
```
?

Reply:
Yes, in this case, we do actually want to ensure that the type is exactly the same, not potentially a subclass. For example, `EnergyoutFilter` is a subclass of `EnergyFilter`, but asking for `EnergyFilter` shouldn't give you indices for an `EnergyoutFilter`.